docs/topics/scripting/custom-script-deps-tutorial.md

```diff
@@ -0,0 +1,330 @@
+
+   ```kotlin
+    // @KotlinScript annotation marks a script definition class
+    @KotlinScript(
+        // File extension for the script type
+        fileExtension = "scriptwithdeps.kts",
+        // Compilation configuration for the script type
+        compilationConfiguration = ScriptWithMavenDepsConfiguration::class
+    )
+    abstract class ScriptWithMavenDeps
+
+    object ScriptWithMavenDepsConfiguration: ScriptCompilationConfiguration()
+   ```
+   
+   > In this tutorial, we provide only the working code without explaining Kotlin scripting API.
+   > You can find the same code with detailed explanation comments [on GitHub](https://github.com/Kotlin/kotlin-script-examples/blob/master/jvm/basic/jvm-maven-deps/script/src/main/kotlin/org/jetbrains/kotlin/script/examples/jvm/resolve/maven/scriptDef.kt).
+   > 
+   {type="note"}
+
+8. Define the script compilation configuration as shown below.
+
+   ```kotlin
+    // ScriptCompilationConfiguration is the base class for compilation configurations
+    object ScriptWithMavenDepsConfiguration : ScriptCompilationConfiguration(
+        {
+            // Implicit imports for all scripts of this type
+            defaultImports(DependsOn::class, Repository::class)
+            jvm {
+                // Extract the whole classpath from context classloader and use it as dependencies
+                dependenciesFromCurrentContext(wholeClasspath = true) 
+            }
+            // Callbacks
+            refineConfiguration {
+                // Process specified annotations with the provided handler
+                onAnnotations(DependsOn::class, Repository::class, handler = ::configureMavenDepsOnAnnotations)
+            }
+        }
+    )
+   ```
+
+   The `configureMavenDepsOnAnnotations` function is as follows:
+
+   ```kotlin
+    // Handler that reconfigures the compilation on the fly
+    fun configureMavenDepsOnAnnotations(context: ScriptConfigurationRefinementContext): ResultWithDiagnostics<ScriptCompilationConfiguration> {
+        val annotations = context.collectedData?.get(ScriptCollectedData.collectedAnnotations)?.takeIf { it.isNotEmpty() }
+            ?: return context.compilationConfiguration.asSuccess()
+        return runBlocking {
+            resolver.resolveFromScriptSourceAnnotations(annotations)
+        }.onSuccess {
+            context.compilationConfiguration.with { 
+                dependencies.append(JvmDependency(it))
+            }.asSuccess()
+        }
+    }
+    
+    private val resolver = CompoundDependenciesResolver(FileSystemDependenciesResolver(), MavenDependenciesResolver())
+   ```
+
+   You can find the full code [here](https://github.com/Kotlin/kotlin-script-examples/blob/master/jvm/basic/jvm-maven-deps/script/src/main/kotlin/org/jetbrains/kotlin/script/examples/jvm/resolve/maven/scriptDef.kt).
+
+
+Now you have the script definition: you know what can be written in the scripts and their file extension. The next step
```

Comment:
I do not understand what you mean by "you know what can be written in the scripts and their file extension".
Maybe "that defines things available for the script authors". I would skip part about the extension here.

Reply:
`that defines things available for the script authors`
This needs some improvements:
- `definition that defines` - repetition
- `things available for the script authors` is too general and lacks precision.
- we also can drop `script definition` here because it repeats multiple times in this subsection.
How about:
`On this step, you have defined what developers can write in scripts of this type and how it will be handled. The next step is... `

Reply:
This sounds good to me, let's use it.